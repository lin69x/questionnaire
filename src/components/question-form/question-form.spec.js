import React from 'react';
import Enzyme  from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {createStore } from 'redux';
import Question from './question-form';

Enzyme.configure({adapter: new EnzymeAdapter()});

const props = {
  headline: 'Headline',
  question_type: 'multiple-choice',
  choices: [
    {
      label: 'option 1',
      value: 'option 1',
    },
    {
      label: 'option 2',
      value: 'option 2',
    }
  ],
  identifier: 'uniqueKey',
  handleSubmit: jest.fn(),
  actions: [
    <button type='submit' key={1}>Test Button</button>
  ],
  multiline: "false",
};

jest.mock('redux-form', () => ({
  Field: 'Field',
    reduxForm: () => component => component,
}));

describe('Question Form', () => {
  it('should match snapshot', () => {
    const renderedValue = renderer.create(<Question {...props}/>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
