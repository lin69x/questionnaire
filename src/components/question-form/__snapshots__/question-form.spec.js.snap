// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Form should match snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div>
    <div
      className="question-container"
    >
      <div>
        <h4>
          Headline
        </h4>
        <label
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <Field
            component="input"
            name="uniqueKey"
            type="radio"
            value="option 1"
          />
           
          option 1
        </label>
        <br />
        <label
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <Field
            component="input"
            name="uniqueKey"
            type="radio"
            value="option 2"
          />
           
          option 2
        </label>
        <br />
      </div>
    </div>
  </div>
  <div>
    <button
      type="submit"
    >
      Test Button
    </button>
  </div>
</form>
`;
